### PR TITLE
worker: replace ginkgo-based tests by testify

### PR DESCRIPTION
**Pull request by cirocosta**
### Why do we need this PR?

moving along with the idea of getting our tests using testify (see
https://github.com/concourse/concourse/issues/4361#issuecomment-531839367),
this commit replaces the initial `ginkgo`-based tests that we had for
`backend` by testify.

note.: although these tests are not using ginkgo, and ginkgo's
`suite_test.go*` is removed, the `ginkgo` command issues by
`concourse/ci` will still get these tests running, although with a
slightly different output (pure `go test` output):

            === RUN   TestSuite
            === RUN   TestSuite/TestPing
            === RUN   TestSuite/TestPing/fail_from_containerd_version_service
            === RUN   TestSuite/TestPing/ok_from_containerd's_version_service
            --- PASS: TestSuite (0.00s)
                --- PASS: TestSuite/TestPing (0.00s)
                    --- PASS: TestSuite/TestPing/fail_from_containerd_version_service (0.00s)
                    --- PASS: TestSuite/TestPing/ok_from_containerd's_version_service (0.00s)
            PASS
            ok      github.com/concourse/concourse/worker/backend

### Changes proposed in this pull request

* use `testify` for `backend_test.go`.

### Contributor Checklist

> Are the following items included as part of this PR? Please delete checkbox items that don't apply.
- [x] Unit tests
  - although this has no new assertions / behavior testing
- ~Integration tests (if applicable)~
  - just refactored unit tests
- ~Updated documentation (located at https://github.com/concourse/docs)~
   - nothing user facing
- ~Updated release notes (located at https://github.com/concourse/concourse/tree/master/release-notes)~
  - nothing user facing

- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the [BOSH](https://github.com/concourse/concourse-bosh-release) 
      and [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for the [integration tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env) (for example, if they are Garden configs that are not displayed in the `--help` text). 

<br />

Thanks!

### CODE REVIEW

File: worker/backend/backend_test.go [hunk 1]
```diff
@@ -2,45 +2,57 @@ package backend_test
...
+		{
+			versionReturn: nil,
+			succeeds:      true,
+		},
+		{
+			versionReturn: errors.New("errr"),
+			succeeds:      false,
+		},
+	} {
+		s.T().Run("case", func(t *testing.T) {
```
Comment (cirocosta) [inline worker/backend/backend_test.go]: this gets expanded to `case#1`, `case#2`, ...

File: go.mod [hunk 2]
```diff
@@ -186,7 +186,7 @@ require (
 	github.com/square/certstrap v1.1.1
 	github.com/streadway/amqp v0.0.0-20190225234609-30f8ed68076e // indirect
 	github.com/stretchr/objx v0.2.0 // indirect
-	github.com/stretchr/testify v1.4.0 // indirect
+	github.com/stretchr/testify v1.4.0
```
Comment (cirocosta) [inline go.mod]: the first `testify`-based tests in our code!

File: worker/backend/backend_test.go (hunk 1 above)
Comment (cirocosta) [reply worker/backend/backend_test.go]: another pattern would be to take `desc` inside the "anonymous struct". wdyt?

Comment (vito) [reply worker/backend/backend_test.go]: yeah the `desc` field pattern seems like a nice idea

Comment (cirocosta) [reply worker/backend/backend_test.go]: looks good indeed 😁  just applied

**Comment by YoussB** [comment]
is there a specific reason we are moving to testify more than it being more widely used?
was looking into it, I found ginkgo better for a couple of reasons:
- ginkgo/gomega tests are way easier to read.
- seems like the ginkgo/gomega codebase is updated more frequently, that might give more stability to that framework 🤷‍♂ 

wdyt?

**Comment by pivotal-jwinters** [comment]
In addition to what @YoussB mentioned, I also feel like it's a greater cognitive burden to have to reason about two different tests suites within the same repo.

**Comment by pivotal-jamie-klassen** [comment]
@YoussB I'd say `JustBeforeEach` is dangerous for readability, but if you do it right ginkgo can be pretty nice. @pivotal-jwinters I think your comment opens a can of worms about how difficult/mixed up our test suite is, and I'm here for it! 🍿

**Comment by vito** [comment]
@YoussB @pivotal-jamie-klassen @pivotal-jwinters The problem with Ginkgo is getting it right. 

* `Describe/Context/It` descriptions are often misleading, misworded, or copy-pasted.
* `Eventually` results in a constant battle with flaky tests.
* `JustBeforeEach` is often misused and confuses people all the time.
* Test setup and assertions are not consistently split between `BeforeEach`/`JustBeforeEach` and `It`.
* Finding where exactly to put a test in an already existing suite can take a lot of time and effort, and it might not end up where others would expect. Over time this can quickly grow into a sea of redundant test trees.

My main interest in `stretchr/testify` isn't just that it's more popular, it's that it's harder to get it *wrong*. It feels like something that would be more intuitive for new contributors, and it feels distinctly more Go-like.

I realize that we have a bunch of existing test suites in Ginkgo. I'm never going to write a story to convert everything over, but while we're embarking on large refactors and new codebases anyway it seemed like a good time to try something that would ease the contributor burden.

**Pull request merged**